### PR TITLE
really upsert in comp_task

### LINKED ISSUE #2125: director-v2 should not re-create the pipeline in comp_tasks but update it
now the director-v2 does re-create the tasks in comp_tasks on each change of the project pipeline.
This works nicely with computational services, because the table is locked from changes while the computational services are running.

But, this is bad with dynamic services because:
- create a jupyter-lab
- generate some big files in the output (like >300MB)
- put them in one output folder
--> this will trigger the node ports and upload the file

- now modifying the project pipeline
--> triggers an update in the project table
--> director-v2 re-creates the computational tasks in comp_tasks
--> when the node ports finishes the upload the task disappeared.
--> the node ports upload completes and the task has been replaced ... effectively it cannot register the data in the table.

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- director-v2 now upserts in comp_tasks table instead of deleting and re-inserting the tasks, so that the tasks remain stable.

## Related issue/s
fixes #2125

### CODE REVIEW

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py [hunk 1]
```diff
@@ -84,72 +148,48 @@ async def _sequentially_upsert_tasks_from_project(
...
-                    else RunningState.NOT_STARTED
-                ),
-                internal_id=internal_id,
-                node_class=node_class,
+        result: RowProxy = await self.connection.execute(
+            sa.select([comp_tasks.c.node_id]).where(
+                comp_tasks.c.project_id == str(project.uuid)
             )
+        )
+        node_id_to_delete = [
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:163]: what happens if the node you are about to delete happens to be running?

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py [hunk 2]
```diff
@@ -84,72 +148,48 @@ async def _sequentially_upsert_tasks_from_project(
...
             )
 
+        # insert or update the remaining tasks
+
+        for comp_task_db in list_of_comp_tasks_in_project:
+
+            insert_stmt = insert(comp_tasks).values(
+                **comp_task_db.dict(by_alias=True, exclude_unset=True)
+            )
+            on_update_stmt = insert_stmt.on_conflict_do_update(
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:185]: updates trigger an event in the webserver, right? Could you please add here a comment of what happens?

**Review by GitHK — APPROVED** [review]
👍

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:163]: The only running nodes are computational. there is a check in /routes/computations.py that checks for that. you cannot modify the comp_tasks table if the computational pipeline is running.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py (hunk 2 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:185]: update triggers only an event if it is the comp_tasks.outputs or comp_tasks.state column being updated. therefore no event is triggered here.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:163]: actually this does not change. the upserting is now done rightfully and before it was done by re-creating the tasks.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py (hunk 2 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:183]: added docs

**Review by pcrespov — APPROVED** [review]

**Pull request merged**